contracts/Tokens/VTokens/VBep20.sol

```diff
@@ -49,7 +49,25 @@ contract VBep20 is VToken, VBep20Interface {
     // @custom:event Emits Transfer event on success
     // @custom:event Emits RedeemFee when fee is charged by the treasury
     function redeem(uint redeemTokens) external returns (uint) {
-        return redeemInternal(redeemTokens);
+        return redeemInternal(msg.sender, msg.sender, redeemTokens);
+    }
+
+    /**
+     * @notice Sender redeems assets on behalf of some other address. This function is only available
+     *   for senders, explicitly marked as delegates of the supplier using `comptroller.updateDelegate`
+     * @dev Accrues interest whether or not the operation succeeds, unless reverted
+     * @param redeemer The user on behalf of whom to redeem
+     * @param redeemTokens The number of vTokens to redeem into underlying
+     * @return uint Returns 0 on success, otherwise returns a failure code (see ErrorReporter.sol for details).
+     */
+    // @custom:event Emits Redeem event on success
+    // @custom:event Emits Transfer event on success
+    // @custom:event Emits RedeemFee when fee is charged by the treasury
+    function redeemBehalf(address redeemer, uint redeemTokens) external returns (uint) {
+        require(comptroller.approvedDelegates(redeemer, msg.sender), "not an approved delegate");
+        address payable receiver = msg.sender;
```

Comment:
Why aren't we doing this (store the `msg.sender` in a payable variable) in the `redeem(uint)` function [here](https://github.com/VenusProtocol/venus-protocol/pull/442/files#diff-9dd0a14d8e80ffd3c6fd54eeeeade7b317e602a2c6f42c28c828bdf23ae765aeR52)?

Reply:
Removing this `payable` conversion as msg.sender below solidity version 0.8 is by default payable and have removed it from other unnecessary places too.

https://github.com/VenusProtocol/venus-protocol/pull/442/commits/c1bcf7a79cf6c64073dc85347c44bc1117ac3c5e